apps/remote_control/test/lexical/remote_control/search/indexer/extractors/function_definition_test.exs

```diff
@@ -147,6 +147,23 @@ defmodule Lexical.RemoteControl.Search.Indexer.Extractors.FunctionDefinitionTest
       assert expected == extract(code, multi_line.range)
     end
 
+    test "finds functions defined in protocol implementations" do
+      {:ok, [function], doc} =
+        ~q[
+          defimpl MyProtocol, for: Structs.Mystruct do
+            def do_proto(a, b) do
+              a + b
+            end
+          end
+        ]
+        |> index_functions()
+
+      assert function.type == :public_function
+      assert function.subject == "MyProtocol.Structs.Mystruct.do_proto/2"
+      assert decorate(doc, function.range) =~ "def «do_proto(a, b)» do"
```

Comment:
This seems to be inconsistent with the other `function`'s definition  range.

Reply:
how so? I think this test is just using decorate rather than extract.